singer-commons/src/main/java/com/pinterest/singer/loggingaudit/client/AuditEventKafkaSender.java

```diff
@@ -216,6 +212,33 @@ private void refreshPartitionIfNeeded() {
     }
   }
 
+  private void resetDefaultPartitionIfNeeded() {
+    if (partitionInfoList.size() == 0) {
+      defaultPartition = 0;
```

Comment:
If partition count is 0 then defaultPartition shouldn't be set to a valid Kafka partition id.

Reply:
the logic behind this is when partitionInfoList.size() is zero,  this means the audit client couldn't get the topic metadata from the Kafka cluster at this moment. metadata being not available could be due to temporal network glitch or Kafka cluster outage.   if we set defaultPartition to be invalid id such as -1,  then the following send() method will ALWAYS fail.  Setting to 0 will increase the chance of successfully sending audit events to Kafka.

Reply:
defaultPartition cannot be 0 if partition size is 0 as that logic is invalid.